packages/postgres-database/src/simcore_postgres_database/migration/versions/35724106de75_new_users_details_table.py

```diff
@@ -0,0 +1,96 @@
+"""new users_details table
+
+Revision ID: 35724106de75
+Revises: 20d60d2663ad
+Create Date: 2024-03-05 13:13:37.921956+00:00
+
+"""
+from typing import Final
+
+import sqlalchemy as sa
+from alembic import op
+
+# revision identifiers, used by Alembic.
+revision = "35724106de75"
+down_revision = "20d60d2663ad"
+branch_labels = None
+depends_on = None
+
+
+# auto-update modified
+# TRIGGERS ------------------------
+_TABLE_NAME: Final[str] = "users_details"
+_TRIGGER_NAME: Final[str] = "trigger_auto_update"  # NOTE: scoped on table
+_PROCEDURE_NAME: Final[
+    str
+] = f"{_TABLE_NAME}_auto_update_modified()"  # NOTE: scoped on database
+
+modified_timestamp_trigger = sa.DDL(
+    f"""
+DROP TRIGGER IF EXISTS {_TRIGGER_NAME} on {_TABLE_NAME};
+CREATE TRIGGER {_TRIGGER_NAME}
+BEFORE INSERT OR UPDATE ON {_TABLE_NAME}
+FOR EACH ROW EXECUTE PROCEDURE {_PROCEDURE_NAME};
+    """
+)
+
+# PROCEDURES ------------------------
+update_modified_timestamp_procedure = sa.DDL(
+    f"""
+CREATE OR REPLACE FUNCTION {_PROCEDURE_NAME}
+RETURNS TRIGGER AS $$
+BEGIN
+  NEW.modified := current_timestamp;
+  RETURN NEW;
+END;
+$$ LANGUAGE plpgsql;
+    """
+)
+
+
+def upgrade():
+    # ### commands auto generated by Alembic - please adjust! ###
+    op.create_table(
+        "users_details",
+        sa.Column("user_id", sa.Integer(), nullable=True),
+        sa.Column("pre_email", sa.String(), nullable=False),
+        sa.Column("pre_first_name", sa.String(), nullable=True),
+        sa.Column("pre_last_name", sa.String(), nullable=True),
+        sa.Column("pre_phone", sa.String(), nullable=True),
+        sa.Column("company_name", sa.String(), nullable=True),
+        sa.Column("address", sa.String(), nullable=True),
+        sa.Column("city", sa.String(), nullable=True),
+        sa.Column("state", sa.String(), nullable=True),
+        sa.Column("country", sa.String(), nullable=True),
+        sa.Column("postal_code", sa.String(), nullable=True),
+        sa.Column("created_by", sa.Integer(), nullable=True),
+        sa.Column(
```

Comment:
question: why are these all nullables? I guess they are not mandatory, but usually they are no?

Reply:
It depends on the configuration of the product. If the product is not billable in reality none of these are mandatory